packages/mobile/src/identity/contactMapping.ts

```diff
@@ -169,31 +170,33 @@ export function* fetchAddressesAndValidateSaga({
     // Clear existing entries for those numbers so our mapping consumers know new status is pending.
     yield put(updateE164PhoneNumberAddresses({ [e164Number]: undefined }, {}))
 
-    const addresses: string[] = yield call(getAddresses, e164Number)
+    const walletAddresses: string[] = yield call(getWalletAddressesAndUpdateCache, e164Number)
 
     const e164NumberToAddressUpdates: E164NumberToAddressType = {}
     const addressToE164NumberUpdates: AddressToE164NumberType = {}
 
-    if (!addresses.length) {
+    if (!walletAddresses.length) {
       Logger.debug(TAG + '@fetchAddressesAndValidate', `No addresses for number`)
       // Save invalid/0 addresses to avoid checking again
       // null means a contact is unverified, whereas undefined means we haven't checked yet
       e164NumberToAddressUpdates[e164Number] = null
     } else {
-      e164NumberToAddressUpdates[e164Number] = addresses
-      addresses.map((a) => (addressToE164NumberUpdates[a] = e164Number))
+      e164NumberToAddressUpdates[e164Number] = walletAddresses
+      walletAddresses.map((a) => (addressToE164NumberUpdates[a] = e164Number))
     }
 
     const userAddress = yield select(currentAccountSelector)
+    const secureSendPossibleAddresses = [...walletAddresses]
     const secureSendPhoneNumberMapping = yield select(secureSendPhoneNumberMappingSelector)
     // If fetch is being done as part of a payment request from an unverified address,
     // the unverified address should be considered in the Secure Send check
-    if (requesterAddress && !addresses.includes(requesterAddress)) {
-      addresses.push(requesterAddress)
+    if (requesterAddress && !secureSendPossibleAddresses.includes(requesterAddress)) {
```

Comment:
I'm not sure I fully understand what's happening here. If we received a payment request, why do we need to do secure send?

Reply:
This is needed to cover an edge case where:
- someone requesting money from you has a verified account on-chain
- but they are making a request from a non-verified account that they own

What this does is just adds the unverified account to the list of possible accounts to disambiguate.